core/lib/generators/spree/install/install_generator.rb

```diff
@@ -83,21 +83,20 @@ def create_overrides_directory
 
     def configure_application
       application <<-RUBY
-    # Load application's model / class decorators
     initializer 'spree.decorators' do |app|
+      # Ensure decorated constants are unloaded before reloading decorators
+      config.to_prepare_blocks.unshift -> { ActiveSupport::Dependencies.clear }
```

Comment:
Will this clear all dependencies declared in all to_prepare blocks of the application? What if the app or other gems added dependencies? Could this be breaking?

Reply:
@tvdeyen that's why I ended up putting it at the beginning with unshift, anyway, although it helps with warnings about redefined constants it doesn't help with the initial problem I was trying to solve.

Digging deeper I've found that when a constant is loaded during the initialization (e.g. inside `config/initializers/_spree.rb`) it won't be reloaded properly and any decorator will end up being prepended at each reload. Aside from that was also evident that `to_prepare` is called twice during reloads, only aggravating the problem.

Right now I'm still experimenting but a partial solution seem to involve a combo of an `initializer` block plus `app.reloader.to_run`, this solves a lot and relies on the call to `ActiveSupport::Dependencies.clear` that rails already does within the reload cycle, but still doens't solve the problem for constants loaded during initialization.

_I'll make sure to mark the PR as ready when I'm more confident that the solution is proven_

Reply:
Thank you so much for looking into this and the detailed explanation. 💯